File: 2021/day1/day1.go
package day1

import (
	"log"
	"strconv"
	"strings"
)

func Run(input string) {
	log.Println("Day 1 Part 2:", Part1(input))
	log.Println("Day 1 Part 2:", Part2(input))
}

func Part2(input string) int {
	lines := strings.Split(input, "\n")

	increases := 0
	for lineNumber, line := range lines {
		intLine, err := strconv.Atoi(line)
		if err != nil {
			log.Fatal(err)
		}

		if lineNumber+3 < len(lines) {
			firstSum := intLine
			secondSum := 0
			for i := 0; i < 3; i += 1 {
				nextLine, err := strconv.Atoi(lines[lineNumber+i+1])
				if err != nil {
					log.Fatal(err)
				}

				if i < 2 {
					firstSum = firstSum + nextLine
				}

				secondSum = secondSum + nextLine
			}

			if secondSum > firstSum {
				increases += 1
			}
		}
	}

	return increases
}

func Part1(input string) int {
	lines := strings.Split(input, "\n")

	increases := 0
	for lineNumber, line := range lines {
		intLine, err := strconv.Atoi(line)
		if err != nil {
			log.Fatal(err)
		}

		if lineNumber+1 < len(lines) {
			nextLine, err := strconv.Atoi(lines[lineNumber+1])
			if err != nil {
				log.Fatal(err)
			}

			if intLine < nextLine {
				increases += 1
			}
		}
	}

	return increases
}

File: 2021/day1/day1.py
# day1.py


def run(input: str):
    print(f"Day 1 Part 1: {part1(input)}")
    print(f"Day 1 Part 2: {part2(input)}")


def part2(input: str):
    increases = 0

    lines = input.splitlines()
    for (line_number, line) in enumerate(lines):
        int_line = int(line)

        if (line_number + 3) < len(lines):
            first_sum = int_line
            second_sum = 0

            for i in range(0, 3):
                next_line = int(lines[line_number + i + 1])

                if i < 2:
                    first_sum += next_line

                second_sum += next_line

            if second_sum > first_sum:
                increases += 1

    return increases


def part1(input: str):
    increases = 0

    lines = input.splitlines()
    for (line_number, line) in enumerate(lines):
        int_line = int(line)

        if (line_number + 1) < len(lines):
            next_int = int(lines[line_number + 1])

            if int_line < next_int:
                increases += 1

    return increases
